src/Traits/HasAddresses.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tipoff\Addresses\Traits;
+
+use Illuminate\Support\Collection;
+use Tipoff\Addresses\Models\Address;
+use Tipoff\Addresses\Models\City;
+use Tipoff\Addresses\Models\DomesticAddress;
+use Tipoff\Addresses\Models\Zip;
+
+/**
+ * @property Collection addresses
+ */
+trait HasAddresses
+{
+    /**
+     * @param string $line1
+     * @param string|null $line2
+     * @param string|City $city
+     * @param string|Zip $zip
+     * @return DomesticAddress
+     */
+    public static function createDomesticAddress(string $line1, ?string $line2, $city, $zip): DomesticAddress
+    {
+        return DomesticAddress::createDomesticAddress($line1, $line2, $city, $zip);
+    }
+
+    public function getAddressByType(string $type): ?Address
+    {
+        return $this->addresses
+            ->filter(function (Address $address) use ($type) {
+                return $address->type === $type;
+            })
+            ->first();
+    }
+
+    public function setAddressByType(string $type, DomesticAddress $domesticAddress): Address
```

Comment:
Setting/saving of the optional fields in the `Address` record is left to the model using this trait.